crates/nu-command/src/platform/input/input_.rs

```diff
@@ -59,114 +68,96 @@ impl Command for Input {
-        // clear terminal events
-        while crossterm::event::poll(Duration::from_secs(0)).map_err(&from_io_error)? {
-            // If there's an event, read it to remove it from the queue
-            let _ = crossterm::event::read().map_err(&from_io_error)?;
-        }
+        let mut line_editor = Reedline::create();
+        line_editor = line_editor.with_ansi_colors(false);
+        line_editor = match history {
+            Some(h) => line_editor.with_history(Box::new(h)),
+            None => line_editor,
+        };
+
+        let mut buf = String::new();
 
         loop {
-            if i64::try_from(buf.len()).unwrap_or(0) >= numchar.item {
-                break;
-            }
-            match crossterm::event::read() {
-                Ok(Event::Key(k)) => match k.kind {
-                    KeyEventKind::Press | KeyEventKind::Repeat => {
-                        match k.code {
-                            // TODO: maintain keycode parity with existing command
-                            KeyCode::Char(c) => {
-                                if k.modifiers == KeyModifiers::ALT
-                                    || k.modifiers == KeyModifiers::CONTROL
-                                {
-                                    if k.modifiers == KeyModifiers::CONTROL && c == 'c' {
-                                        crossterm::terminal::disable_raw_mode()
-                                            .map_err(&from_io_error)?;
-                                        return Err(IoError::new(
-                                            std::io::ErrorKind::Interrupted,
-                                            call.head,
-                                            None,
-                                        )
-                                        .into());
-                                    }
-                                    continue;
-                                }
-
-                                if let Some(bytes_until) = bytes_until.as_ref() {
-                                    if bytes_until.bytes().contains(&(c as u8)) {
-                                        break;
-                                    }
-                                }
-                                buf.push(c);
-                            }
-                            KeyCode::Backspace => {
-                                let _ = buf.pop();
-                            }
-                            KeyCode::Enter => {
-                                break;
-                            }
-                            _ => continue,
-                        }
-                    }
-                    _ => continue,
-                },
+            match line_editor.read_line(&prompt) {
+                Ok(Signal::Success(buffer)) => {
+                    buf.push_str(&buffer);
+                    break;
+                }
+                Ok(Signal::CtrlC) => {
```

Comment:
I think `Signal::CtrlD` should also be handled,

Reply:
:+1: